decode_test.go

```diff
@@ -1456,9 +1496,30 @@ func (ft *obsoleteFailingUnmarshaler) UnmarshalYAML(unmarshal func(interface{})
 }
 
 func TestObsoleteUnmarshalerError(t *testing.T) {
-	err := yaml.Unmarshal([]byte("a: b"), &obsoleteFailingUnmarshaler{})
-	if err != failingErr {
-		t.Fatalf("Unmarshal() returned %v, want %v", err, failingErr)
+	data := `{foo: 123, bar: {}, spam: "test"}`
+	dst := struct {
+		Foo  int
+		Bar  *obsoleteFailingUnmarshaler
+		Spam string
+	}{}
+	err := yaml.Unmarshal([]byte(data), &dst)
+	expectedErr := &yaml.TypeError{
+		Errors: []*yaml.UnmarshalError{
+			{Line: 1, Column: 17, Err: failingErr},
+		},
+	}
+	if !reflect.DeepEqual(expectedErr, err) {
+		t.Errorf("expected error %#v, got #%v", expectedErr, err)
+	}
+	// whatever could be unmarshaled must be unmarshaled
+	if got, want := dst.Foo, 123; got != want {
+		t.Errorf("foo unmarshaled incorrectly, got %#v, want  %#v", got, want)
+	}
+	if got, want := dst.Bar, (&obsoleteFailingUnmarshaler{}); !reflect.DeepEqual(got, want) {
+		t.Errorf("bar unmarshaled incorrectly, got %#v, want  %#v", got, want)
+	}
+	if got, want := dst.Spam, "test"; got != want {
+		t.Errorf("spam unmarshaled incorrectly, got %#v, want  %#v", got, want)
```

Comment:
Remove the extra spaces


```suggestion
		t.Errorf("foo unmarshaled incorrectly, got %#v, want %#v", got, want)
	}
	if got, want := dst.Bar, (&obsoleteFailingUnmarshaler{}); !reflect.DeepEqual(got, want) {
		t.Errorf("bar unmarshaled incorrectly, got %#v, want %#v", got, want)
	}
	if got, want := dst.Spam, "test"; got != want {
		t.Errorf("spam unmarshaled incorrectly, got %#v, want %#v", got, want)
```